code/modules/cm_marines/Xeno_Research/Xeno_Analyzation.dm

```diff
@@ -0,0 +1,257 @@
+				else
+					new /obj/item/marineResearch/xenomorph/chitin(linked_chamber.loc)
+
+				new /obj/item/marineResearch/xenomorph/muscle(linked_chamber.loc)
+
+				switch(linked_chamber.occupant.caste_base_type)
+
+					if(/mob/living/carbon/Xenomorph/Warrior)
+						new /obj/item/marineResearch/xenomorph/muscle(linked_chamber.loc)
+
+					if(/mob/living/carbon/Xenomorph/Ravager)
+						new /obj/item/marineResearch/xenomorph/muscle(linked_chamber.loc)
+
+					if(/mob/living/carbon/Xenomorph/Drone)
+						new /obj/item/marineResearch/xenomorph/secretor(linked_chamber.loc)
+
+					if(/mob/living/carbon/Xenomorph/Sentinel)
+						new /obj/item/marineResearch/xenomorph/acid_gland(linked_chamber.loc)
+
+					if(/mob/living/carbon/Xenomorph/Spitter)
+						new /obj/item/marineResearch/xenomorph/acid_gland/spitter(linked_chamber.loc)
+
+					if(/mob/living/carbon/Xenomorph/Hivelord)
+						new /obj/item/marineResearch/xenomorph/secretor(linked_chamber.loc)
+						new /obj/item/marineResearch/xenomorph/secretor/hivelord(linked_chamber.loc)
+						new /obj/item/marineResearch/xenomorph/secretor/hivelord(linked_chamber.loc)
+
+					if(/mob/living/carbon/Xenomorph/Queen)
+						new /obj/item/marineResearch/xenomorph/acid_gland/spitter(linked_chamber.loc)
+						new /obj/item/marineResearch/xenomorph/secretor(linked_chamber.loc)
+						new /obj/item/marineResearch/xenomorph/secretor/hivelord(linked_chamber.loc)
+						new /obj/item/marineResearch/xenomorph/core(linked_chamber.loc)
+
+				updateUsrDialog()
+
+	updateUsrDialog()
+
+/obj/machinery/computer/analyze_console/attack_hand(mob/user as mob)
+	if(stat & (BROKEN|NOPOWER))
+		return
+
+	user.set_interaction(src)
+	var/dat = ""
+	switch(screen) //A quick check to make sure you get the right screen when a device is disconnected.
+		if(2 to 2.9)
+			if(linked_chamber == null)
+				screen = 2.0
+			else if(linked_chamber.occupant == null)
+				screen = 2.1
+			else if(linked_chamber.occupant.autopsied == 1)
+				screen = 2.3
+			else
+				screen = 2.2
+
+	switch(screen)
+
+		if(0.0)
+			dat += "Establishing linkage with nearby chamber"
+
+		if(0.1)
+			dat += "Scanning in progress."
+
+		if(0.2)
+			dat += "SYSTEM LOCKED<BR><BR>"
+			dat += "<A href='?src=\ref[src];lock=1.2'>Unlock</A>"
+
+		if(0.3)
+			dat += "Harvesting...<BR><BR>"
+
+		if(1.0)
+			dat += "Main Menu:<BR><BR>"
+			if(linked_chamber != null) dat += "<A href='?src=\ref[src];menu=2.2'>Analyze Chamber Menu</A><BR>"
+			else dat += "NO CHAMBER LINKED<BR>"
+			dat += "<HR><A href='?src=\ref[src];menu=1.2'>Settings</A>"
+
+		if(1.2)
+			dat += "<A href='?src=\ref[src];menu=1.0'>Main Menu</A><HR>"
+			dat += "R&D Console Setting:<BR><BR>"
+			dat += "<A href='?src=\ref[src];menu=1.3'>Device Linkage Menu</A><BR>"
+			dat += "<A href='?src=\ref[src];lock=0.2'>Lock Console</A><BR>"
+
+		if(1.3)
+			dat += "<A href='?src=\ref[src];menu=1.0'>Main Menu</A> || <A href='?src=\ref[src];menu=1.2'>Settings</A><HR>"
```

Comment:
You should use defines (example: [MARINE_RESEARCH_MENU_MAIN]) for the h_ref menu values. It makes reading it easier.